### ISSUE TITLE
The duration 28 days and 1 week, is not even that long

### ISSUE THREAD

**Issue by Anthrwpos** [Helioviewer-Project/helioviewer.org#361]
(I think it's not an issue, but i want to ask)
Hi, why the duration of 28 days (actually is 14 days) and 1 week (actually is 3 and half days), it's not even that long in the videos results?
Thanks

**Comment by dgarciabriseno** [comment]
When you create a movie using the duration dropdown, it splits the time evenly before and after the current observation date.

So if I'm looking at a movie on 2022-05-15 and choose 28 days, then it will generate a movie from 2022-05-01 to 2022-05-29, 14 days back and 14 days forward.

**Comment by Anthrwpos** [comment]
Hi, to be honest, i don't see answer to the question, or i didn't understand. If "_it splits the time evenly before and after the current observation date._", then, why doesn't write "14 days"? About "_then it will generate a movie from 2022-05-01 to 2022-05-29, 14 days back and 14 days forward._" but it's 14 days, not 1 month. I don't get it.
Thanks

**Comment by dgarciabriseno** [comment]
You are selecting the *duration* of the video.
The current display in the viewport is chosen as the middle of the video.

In the example we're discussing, when you select a duration of 28 days, then `14 days back + 14 days forward = 28 days`.

**Comment by Anthrwpos** [comment]
> You are selecting the _duration_ of the video. The current display in the viewport is chosen as the middle of the video.
> 
> In the example we're discussing, when you select a duration of 28 days, then `14 days back + 14 days forward = 28 days`.

I can't understand the "14 days back + 14 days forward = 28 days". In [this ](https://helioviewer.org/?movieId=Tz8N5)video, the duration is from 7/6/2022 to 21/6/2022= 14 days, where in this is 14 days back + 14 days forward? It's not like "28 Earths days"?
Thanks

**Comment by dgarciabriseno** [comment]
So you are saying that you went to generate a movie with duration set to 28 days.

![Observation date 2022-06-23, Duration 28 days](https://user-images.githubusercontent.com/94071409/176753316-72394d21-fcf1-4028-aaf3-e8968fdd2441.png)

And the resulting movie was not over a 28 day range?

![image](https://user-images.githubusercontent.com/94071409/176753539-51037151-101b-49a8-9ea6-11a59dfb0346.png)

Here my observation date was 2022-06-23 and the resulting video did request a 28 day range.

**Comment by dgarciabriseno** [comment]
Is this behavior consistent?

I do see a potential way this could happen
If you go into advanced settings, change the time, go back to less settings, then click ok to submit the movie. Then time range that was set in advanced settings will be used instead of the selected duration.

@btaylor-sesda Can you fix this? The time range needs to be updated when you click "Less Settings" to whatever the selected duration is.

**Comment by Anthrwpos** [comment]
Yes, the result is not 28 days, it's 14. In your observation date, did you saw the video and is 28 days or only you show me this image?

**Comment by dgarciabriseno** [comment]
Ah I see. Sorry, I was only looking at the request times, not the actual result times.
I understand the problem now. The video is not 28 days. And any video made when observing today's date will not be 28 days. It will be like you said, half the time.
The reason it is only half the time is because like I said, Helioviewer will request a video that is 14 days behind and 14 ahead of the current observation date. If you are looking at today's date, Helioviewer can't get images 14 days into the future, so that portion of the range is ignored, and the result is a video that is only 14 days behind.

As a real example, for the video you shared, here's what's in the database.

```
+---------------------+---------------------+---------------------+---------------------+---------------------+
| reqObservationDate  | reqStartDate        | reqEndDate          | startDate           | endDate             |
+---------------------+---------------------+---------------------+---------------------+---------------------+
| 2022-06-21 21:10:05 | 2022-06-07 21:10:05 | 2022-07-05 21:10:05 | 2022-06-07 21:10:05 | 2022-06-21 19:58:17 |
+---------------------+---------------------+---------------------+---------------------+---------------------+
1 row in set (0.00 sec)

+---------------------+
| modified            |
+---------------------+
| 2022-06-27 18:14:34 |
| 2022-06-27 18:14:34 |
+---------------------+
```

This means:
- The day and time you made the movie (or rather, when Helioviewer completed the movie you requested) was 2022-06-27 18:14:34.
- The UI had an observation date of 2022-06-21 21:10:05.
- A 28 day movie was requested, 14 days back, and 14 days forward means the requested start & end times were 2022-06-07 21:10:05 to 2022-07-05 21:10:05.
- At the time you made the request, there were no images available after 2022-06-21 19:58:17 due to the power outage at Stanford, and the delay in me retrieving images after they came back online.
- The result: you end up with a video that is only 14 days long, instead of 28 days.

This is one of those "it's not a bug, it's a feature" situations... For now the workaround would be to set your date range explicitly in advanced settings.

@wafels I propose we add a slider or something to say where the current observation date should appear in the video, beginning, middle, or end. (or custom % if you want to be fancy)

**Comment by wafels** [comment]
I like the idea of setting "beginning, middle or end" as long as the default is "middle". Maybe this can go in 'Advanced Settings'?

**Comment by dgarciabriseno** [comment]
Advanced Settings let's you set the start and end time, I don't think it needs any other options. For the simple settings beginning/middle/end would be good.

**Comment by Anthrwpos** [comment]
This [video](https://helioviewer.org/?movieId=ML8N5) is requested as 28 days but it came as 14 days again, even if there is available data for each day.
I still don't understand, what do you mean about "14 days back, and 14 days forward", it doesn't make sense for me, from the moment I apply for 28 days and not for 14 days. 
Sorry for repeating this topic, but I would like to understand it better if you don't mind. 

Thanks

**Comment by dgarciabriseno** [comment]
Actually, right now for EUVI, 2022-06-28 23:57:00 is the latest available images.

"14 days back, and 14 days forward" from the observation date means starting with the current image you're looking at in the viewport, that image should the exact middle of the video that gets created.

See [this video](https://helioviewer.org/?movieId=m48N5) I created with this [viewport](https://bit.ly/3bRSPsr). Click that link to get the same viewport I used to create the 28 day movie, the result will be 28 days.

**Comment by Anthrwpos** [comment]
So, what it matters, is the viewport? I mean, if i 'create' 2 videos, 1 from far and the other from close up, the resault from far will be 28 days because it collect more data, but from close up, no, becasuse from the spesific viewpoint, maybe it has not the same amount of data? Correct?

But [here](https://helioviewer.org/?date=2022-06-28T23:55:45.000Z&imageScale=4.84088176&centerX=-96.81763520000001&centerY=-58.09058112&imageLayers=%5BSTEREO_A,SECCHI,EUVI,304,1,100,0,60,1,2022-05-25T13:59:14.000Z.000Z%5D&eventLayers=&eventLabels=true&celestialBodies=%7B%22soho%22:%5B%22soho-mercury-tree-label%22,%22soho-uranus-tree-label%22%5D,%22stereo_a%22:%5B%5D,%22stereo_b%22:%5B%5D%7D) (viewpoint), data before 28/6/22, is available, but it creates 14 days [again](https://helioviewer.org/?movieId=h5KN5).

Thanks

**Comment by dgarciabriseno** [comment]
> But [here](https://helioviewer.org/?date=2022-06-28T23:55:45.000Z&imageScale=4.84088176&centerX=-96.81763520000001&centerY=-58.09058112&imageLayers=%5BSTEREO_A,SECCHI,EUVI,304,1,100,0,60,1,2022-05-25T13:59:14.000Z.000Z%5D&eventLayers=&eventLabels=true&celestialBodies=%7B%22soho%22:%5B%22soho-mercury-tree-label%22,%22soho-uranus-tree-label%22%5D,%22stereo_a%22:%5B%5D,%22stereo_b%22:%5B%5D%7D) (viewpoint), data before 28/6/22, is available, but it creates 14 days [again](https://helioviewer.org/?movieId=h5KN5).

data before 28/6/22 is available, but data after is not.

It's not how far or close up. It is the *date* set in the viewport.

Here, the latest data is 28/6/22
![image](https://user-images.githubusercontent.com/94071409/177542389-e9c16d76-48cf-4b3a-a203-a4a44f1af220.png)

When requesting a video, it attempts to put 28/6/22 in the middle of the video.
The video helioviewer tries to create when the *observation date* is 28/6/22 is:

```
| 14/6/22 --------------- 28/6/22 --------------- 12/7/22 |
14 days earlier         Viewport Date            14 days after
```

For `EUVI-A 304` there is nothing available after 28/6/22, so the video ends there.

If you change the date to 14/6/22
![image](https://user-images.githubusercontent.com/94071409/177543174-5e5ca9d5-2c5d-4b1b-b4be-32e87e703a97.png)

Then the full video will be
```
| 31/5/22 --------------- 14/6/22 --------------- 28/6/22 |
14 days earlier           Viewport Date               14 days after
```

**Comment by Anthrwpos** [comment]
I think i understand from your latest reply. It's, if one day from entire video (and same from other days), have some hours missing (?).
Thanks for your time and sorry.
